src/inspect_ai/agent/_as_solver.py

```diff
@@ -52,10 +55,13 @@ def as_solver(agent: Agent, **agent_kwargs: Any) -> Solver:
     @solver(name=agent_name)
     def agent_to_solver() -> Solver:
         async def solve(state: TaskState, generate: Generate) -> TaskState:
-            # run agent
-            agent_state = await agent(
-                AgentState(messages=state.messages), **agent_kwargs
-            )
+            # run the agent with limits
+            agent_state = AgentState(messages=state.messages)
+            try:
+                with using_limits(limits):
```

Comment:
Now I see the a possible problem with `limits()` -- would it conflict w/ the `limits` variable name above? I think this could be fine as we could just alias the function name but anyone doing something similar would need to know to do this as well. Let's discuss realtime. We could consider `use_limits()` (so as to raise our C# radar!) or `apply_limits()`. I prefer `limits()` but let's talk this through.